determined-ai/determined
# feat: allow users with role Viewer and above to view resource quotas

## Ticket

DET-10430


## Description

Allowed users with  Viewer, Editor, Editor Restricted, Editor Project Restricted, Gen AI and Workspace Admin to be able to view resource quotas in the WebUI when RBAC is enabled. Cluster Admin can still modify and view resource quotas. Allowed all users to view quotas when RBAC is disabled. This involved allowing users who cannot modify the workspaces to also see the Edit Workspace Modal but they will be seeing a "Config" option instead of "Edit Workspace". 


## Test Plan

Set up minikube and create a namespace "test-ns" with a resource quota of 2.
- Login as Cluster Admin and create a workspace "w1" with no namespace binding, "w2" with auto-created namespace binding and some resource quota. and "w3" with "test-ns" as the namespace.
- Edit "w1" and change the binding to an auto-created namespace with some resource quota. Make sure it saves correctly.
- Edit "w2" and change the binding to "test-ns". Make sure that the updated resource quota shows up as 2.
- Edit "w3" and change the binding to an auto-created namespace with a resource quota of 2, and make sure that it saves correctly.
- login with Workspace Admin, and try to make changes to some of the workspaces and make sure that resource quota is not editable and there are no errors while saving the workspaces.
-  log in as workspace creator and make sure that viewing resource quota is non visible.
- log in with the rest of the roles and make sure that resource quotas show up correctly and noneditable for:

1. Workspace with no binding
2. workspace with binding but no quota
3. workspace with autocreated binding and quota
4. workspace with other namespace binding and quota

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.




[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
